packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx

```diff
@@ -0,0 +1,200 @@
+import * as Ariakit from '@ariakit/react';
+import clsx from 'clsx';
+import * as React from 'react';
+import { ScrollAreaV2 } from 'ui-design-system';
+
+const WithComboboxContext = React.createContext(false);
+const WithComboboxListContext = React.createContext(false);
+const ParentContext = React.createContext(false);
+const HideAllContext = React.createContext<(() => void) | null>(null);
+
+function HideAllContextProvider({ children }: { children: React.ReactNode }) {
+  const hideAll = Ariakit.useMenuContext()?.hideAll ?? null;
+  return (
+    <HideAllContext.Provider value={hideAll}>
+      {children}
+    </HideAllContext.Provider>
+  );
+}
+
+export interface MenuRootProps {
+  searchValue?: string;
+  onSearch?: (value: string) => void;
+  children: React.ReactNode;
+}
+
+/**
+ * A menu component.
+ *
+ * @example
+ *  <MenuRoot>
+ *    <MenuButton />
+ *    <MenuPopover>
+ *      <MenuCombobox />
+ *      <MenuContent>
+ *        <MenuItem />
+ *        <MenuItem />
+ *        <MenuSeparator />
+ *        <MenuGroup>
+ *          <MenuGroupLabel />
+ *          <MenuItem />
+ *          <MenuItem />
+ *        </MenuGroup>
+ *      </MenuContent>
+ *    </MenuPopover>
+ *  </MenuRoot>
+ */
+export function MenuRoot({ children, searchValue, onSearch }: MenuRootProps) {
+  const parent = React.useContext(ParentContext);
+  const withCombobox = searchValue !== undefined || onSearch !== undefined;
+
+  const element = (
+    <Ariakit.MenuProvider
+      showTimeout={0}
+      placement={parent ? 'right-start' : 'bottom-start'}
+    >
+      <WithComboboxContext.Provider value={withCombobox}>
+        <HideAllContextProvider>{children}</HideAllContextProvider>
```

Comment:
what does this do exactly? Hide context providers from screen readers ?

Reply:
This is a context I created in order to inject a hideAll method.
This allows to close all menu and submenu at once